### PR TITLE
chore: Add common steps to utilities

### LINKED ISSUE #207: Create verify step to check build time
We need to verify that S2I build time is consistently getting under one minute mark. Also on top of that check Maven build log to verify that no external repositories are used and we fully leverage the S2I image with dependencies included.

### PR DESCRIPTION
**Pull request by dsimansk**
Resolves #207 

This PR:
- creates new sub-package `bdd` in utilities to accommodate commons steps 
- changes the BDD binding to directive style (seems more natural as this action is not performed by "user" directly)
- adds `verifyBuild method`
  - `@Then("^verify s2i build of integration \"([^\"]*)\" was finished in duration (\\d+) min$")`

### CODE REVIEW

File: utilities/src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java [hunk 1]
```diff
@@ -22,8 +22,9 @@
 @Slf4j
 public class SyndesisTemplate {
 
-	private static final String SUPPORT_SA_URL = "https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/support/serviceaccount-as-oauthclient-restricted.yml";
-	private static final String TEMPLATE_URL = "https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/syndesis-restricted.yml";
+	//TODO: move this to TestConfiguration
+	private static final String SUPPORT_SA_URL = "https://raw.githubusercontent.com/syndesisio/syndesis/master/deploy/support/serviceaccount-as-oauthclient-restricted.yml";
```
Comment (tplevko) [inline utilities/src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java]: fix the URL

File: utilities/src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java [hunk 2]
```diff
@@ -22,8 +22,9 @@
...
+	private static final String TEMPLATE_URL = "https://raw.githubusercontent.com/syndesisio/syndesis/master/deploy/syndesis-restricted.yml";
```
Comment (tplevko) [inline utilities/src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java]: fix URL

File: ui-tests/src/test/java/io/syndesis/qe/CucumberTest.java [hunk 3]
```diff
@@ -1,32 +1,25 @@
...
 
 import cucumber.api.CucumberOptions;
 import cucumber.api.junit.Cucumber;
 
 @RunWith(Cucumber.class)
 @CucumberOptions(
 		features = "classpath:features",
-		format = {"pretty"}
+		format = {"pretty", "html:target/cucumber-report.html", "junit:target/cucumber-junit.html"}
 )
```
Comment (tplevko) [inline ui-tests/src/test/java/io/syndesis/qe/CucumberTest.java]: I needed some glue here: glue = {"io.syndesis.qe.bdd"})

**Comment by dsimansk** [comment]
@tplevko I've fixed those changed URLs and also made them more configurable. The second issue with missing glue is also fixed. As we've discussed on IRC might be worth revisit the package model to make the common steps auto-discoverable maybe?

File: utilities/src/main/java/io/syndesis/qe/TestConfiguration.java [hunk 4]
```diff
@@ -29,6 +29,9 @@
 	public static final String SYNDESIS_CREDENTIALS_FILE = "syndesis.config.credentials.file";
 	public static final String SYNDESIS_VERSIONS_FILE = "syndesis.config.versions.file";
 
+	public static final String SYNDESIS_TEMPLATE_URL = "syndesis.config.template.url";
+	public static final String SYNDESIS_TEMPLATE_SA = "syndesis.config.template.sa.";;
```
Comment (tplevko) [inline utilities/src/main/java/io/syndesis/qe/TestConfiguration.java]: cosmetic issue: ";;" and "syndesis.config.template.sa." - remove dot from the end

**Comment by dsimansk** [comment]
@tplevko all done hopefully.

**Pull request merged**